state.go

```diff
@@ -0,0 +1,16 @@
+package specs
+
+// State holds information about the runtime state of the container.
+// This information will be stored in a file called `state.json`.
+// The location of this file will be operating system specific. On Linux
+// it will be in `/run/opencontainers/runc/<containerID>/state.json`
+type State struct {
+	// Version is the version of the specification that is supported.
```

Comment:
I've [wanted `state.go` in its own file][1] for a while now, but:

* It seems like an independent issue from the thrust of this PR, and
* Previous pushback revolved around [limiting the number of `*.go` files][2]

[1]: https://github.com/opencontainers/specs/pull/87#issuecomment-137200444
[2]: https://github.com/opencontainers/specs/pull/87#issuecomment-128137602